jaypeasee/sorting-hat-quiz
# Refactor - Data Management

#### What’s this PR do?
This PR refactors under the hood of the app to manage data better. This includes:
1. Creating a utilities file to clean fetched API data before it is assigned to state.
2. Implements propTypes on all components for type checking throughout the page.
3. Creates a loading message for character data in `Result` component before they are officially rendered.

#### Where should the reviewer start?
1. Review every component except App to see how propTypes are implemented on the page.
2. Review `Result` component render to see the loading message for characters.
3. Review `utilities.js` and the functions that invoke it to see how data is now cleaned.

#### How should this be manually tested?
For checking propTypes:
1. Run `npm start`
2. Open the console in dev tools.
3. Take the quiz and make sure there are no type warnings in the console.

For `utilties.js`:
1. Run `npm start`
2. Make sure App still updates state on load and that all the same house data is present except `_id`.
3. Get to the `Result` and confirm that state updates `houseMates` with only `name` and `_id` of the characters in the corresponding house result.

For the loading message:
1. Run `npm start`
2. Get to the `Result` and confirm characters still load.

#### What are the relevant tickets?
* #36 
* #35 
* #34
